src/api/controller/api/publish.js

```diff
@@ -14,15 +13,32 @@ export const publishMiddleware = async (ctx) => {
 
     const columns = await ctx.field.findAll();
 
-    const transformDocument = ctx.getDocumentTransformer(columns);
+    const transformDocumentFactory = ctx.getDocumentTransformer({
+        getDocumentByField: ctx.publishedDataset.findOneByField,
+        naan: ctx.ezMasterConfig.naan,
+        subpublisher: ctx.ezMasterConfig.subpublisher,
+    })(columns);
 
+    const transformDocumentPreStage = transformDocumentFactory('pre');
+    let handled = 0;
     while (handled < count) {
         const dataset = await ctx.dataset.findLimitFromSkip(100, handled);
-        const transformedDataset = await Promise.all(dataset.map(transformDocument));
+        const transformedDataset = await Promise.all(dataset.map(transformDocumentPreStage));
         await ctx.publishedDataset.insertMany(transformedDataset);
         handled += dataset.length;
     }
 
+    const transformDocumentPostStage = transformDocumentFactory('post');
+    handled = 0;
+
+    while (handled < count) {
```

Comment:
Should'nt the uri be added at the start, and then th Link operation could be executed at the same time as all the other.